--- origsrc/libxmp-4.4.0/Makefile.in	2016-07-16 06:29:48.000000000 -0500
+++ src/libxmp-4.4.0/Makefile.in	2016-08-08 17:09:39.754092700 -0500
@@ -35,6 +35,7 @@ SOLIB		= libxmp.so
 SHLIB		= $(SOLIB).$(VERSION)
 SONAME		= $(SOLIB).$(VERSION_MAJOR)
 DLL		= libxmp.dll
+CYGDLL		= cygxmp-$(VERSION_MAJOR).dll
 DYLIB		= libxmp.$(VERSION_MAJOR).dylib
 GCLIB		= libxmp-gc.a
 
@@ -80,6 +81,8 @@ coverage: lib/$(GCLIB)
 
 dll: lib/$(DLL)
 
+cygdll: lib/$(CYGDLL)
+
 dylib: lib/$(DYLIB)
 
 lib/$(LIB): $(OBJS)
@@ -103,6 +106,12 @@ lib/$(DLL): $(OBJS)
 	if [ "$(V)" -gt 0 ]; then echo $$CMD; else echo LD $@ ; fi; \
 	eval $$CMD
 
+lib/$(CYGDLL): $(OBJS)
+	@mkdir -p lib
+	@CMD='$(LD) $(LDFLAGS) -shared -Wl,--out-implib,lib/libxmp.dll.a -o $@ $(OBJS) $(LIBS)'; \
+	if [ "$(V)" -gt 0 ]; then echo $$CMD; else echo LD $@ ; fi; \
+	eval $$CMD
+
 # From http://stackoverflow.com/questions/15905310:
 # The version number checks that dyld performs are limited to ensuring that
 # the compatibility version of the library being loaded is higher than the
@@ -132,6 +141,12 @@ install: all
 		echo "Installing $(LIB)..."; \
 		$(INSTALL) -m644 lib/$(LIB) $(DESTDIR)$(LIBDIR); \
 	fi
+	@if [ -f lib/$(CYGDLL) ]; then \
+		echo "Installing $(CYGDLL)..."; \
+		mkdir -p $(DESTDIR)$(BINDIR); \
+		$(INSTALL) -m755 lib/$(CYGDLL) $(DESTDIR)$(BINDIR); \
+		$(INSTALL) -m644 lib/libxmp.dll.a $(DESTDIR)$(LIBDIR); \
+	fi
 	@if [ -f lib/$(DYLIB) ]; then \
 		echo "Installing $(DYLIB)..."; \
 		$(INSTALL) -m755 lib/$(DYLIB) $(DESTDIR)$(LIBDIR); \
--- origsrc/libxmp-4.4.0/configure.ac	2016-04-22 20:39:30.000000000 -0500
+++ src/libxmp-4.4.0/configure.ac	2016-08-08 17:09:39.754092700 -0500
@@ -40,7 +40,7 @@ powerpc64)
 esac
 
 case "${host_os}" in
-cygwin* | mint*)
+mint*)
   enable_static=yes
   enable_shared=no
   ;;
@@ -57,7 +57,11 @@ if test "${enable_shared}" != no; then
     AC_CHECK_DEFINED(__APPLE__,[
       AC_SUBST(SHARED,dylib)
     ],[
-      AC_SUBST(SHARED,shared)
+      AC_CHECK_DEFINED(__CYGWIN__,[
+        AC_SUBST(SHARED,cygdll)
+      ],[
+        AC_SUBST(SHARED,shared)
+      ])
     ])
   ])
 fi
--- origsrc/libxmp-4.4.0/src/common.h	2016-07-16 06:37:36.000000000 -0500
+++ src/libxmp-4.4.0/src/common.h	2016-08-08 17:09:39.754092700 -0500
@@ -10,7 +10,7 @@
 #include "xmp.h"
 
 #if defined(__GNUC__) || defined(__clang__)
-#if !defined(WIN32) && !defined(ANDROID) && !defined(__APPLE__) && !defined(__AMIGA__) && !defined(B_BEOS_VERSION) && !defined(__ATHEOS__) && !defined(EMSCRIPTEN) && !defined(__MINT__) 
+#if !defined(WIN32) && !defined(ANDROID) && !defined(__APPLE__) && !defined(__AMIGA__) && !defined(B_BEOS_VERSION) && !defined(__ATHEOS__) && !defined(EMSCRIPTEN) && !defined(__MINT__) && !defined(__CYGWIN__)
 #define USE_VERSIONED_SYMBOLS
 #endif
 #endif
